File: app/src/main/java/com/mdgd/pokemon/models/repo/network/Network.java
package com.mdgd.pokemon.models.repo.network;

import com.mdgd.pokemon.models.infra.Result;
import com.mdgd.pokemon.models.repo.network.schemas.PokemonDetails;

import java.util.List;

import io.reactivex.rxjava3.core.Observable;
import io.reactivex.rxjava3.core.Single;

public interface Network {
    Observable<Result<List<PokemonDetails>>> loadPokemons(int bulkSize);

    Single<Result<List<PokemonDetails>>> loadPokemons(Integer page, int pageSize);

    Single<Result<Long>> getPokemonsCount();
}

File: models/src/main/java/com/mdgd/pokemon/models/repo/network/Network.kt
package com.mdgd.pokemon.models.repo.network

import com.mdgd.pokemon.models.repo.network.schemas.PokemonDetails

interface Network {
    suspend fun loadPokemons(pokemonsCount: Long, offset: Long): List<PokemonDetails>
    suspend fun loadPokemons(page: Int, pageSize: Int): List<PokemonDetails>
    suspend fun getPokemonsCount(): Long
}
